### PR TITLE
chess_onchain

### PR DESCRIPTION
**Pull request by Teke-pixel**
#### Project Abstract

> Please replace these instructions with a brief description of your project summarising key points (1-2 paragraphs).
>
> If your application is a follow-up to a previous grant, please mention which one in the first line of the abstract and include a link to previous pull requests if applicable.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`project_name.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] Payment details have been provided (bank details via email _or_ BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (USDT) address in the application).
- [ ] The software delivered for this grant will be released under an open-source license specified in the application.
- [ ] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [ ] I prefer the discussion of this application to take place in a private Element/Matrix channel. My username is: `@_______:matrix.org` (change the homeserver if you use a different one)

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for your interest in our grants program. But you are currently trying to add two files as part of this PR to our repo. Could you only commit one application? Also, the .md ending is missing.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Also, see the default deliverables of our template: https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md#milestone-1-example--basic-functionality

**Comment by Teke-pixel** [comment]
Ok , I'll update , thanks for reaching out

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the quick reply here. Let me know if I should take another look at the application.

**Comment by Teke-pixel** [comment]
Not yet I’m doing the proper milestones documentation 👍👍 I’ll send a
follow up when I’m done ✔️

**Review by semuelle — CHANGES_REQUESTED** [review]
Thanks for the application, @Teke-pixel. There are some issues that I've listed below, but mainly I am concerned about the lack of details about your platform as a whole. https://github.com/InCryptoEncrypted/ChessOnChain is either private or does not exist, and there are so many keywords in this application that have no technical details anywhere (governance, multichain).

- You have listed 15k as the total cost, but two milestones with 15k each. 
- M2 doesn't have a list of deliverables. Incorporation is not covered by the grants program. See our [FAQ](https://w3f.github.io/Grants-Program/docs/faq#what-activitiespositions-do-you-fund).
- You mention a Milestone 3, but nothing is listed 

I think chess on chain is a fun idea, but this might be better suited to be implemented on top of the various gaming platforms/libraries that are already being built or perhaps as smart contracts. For example, Ajuna has a generic board game pallet.

**Comment by semuelle** [comment]
I am closing this PR due to inactivity. Let me know if you want to update it, I'll reopen it then.

**Comment by Teke-pixel** [comment]
> I am closing this PR due to inactivity. Let me know if you want to update it, I'll reopen it then.

hello semuelle pls reopen , i took my time to define technical details .

**Comment by Noc2** [comment]
I reopened the PR.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. But the milestone tables of the template are still not part of the application. Feel free to ping me once I should take another look at it.

**Comment by Teke-pixel** [comment]
> Thanks for the update. But the milestone tables of the template are still not part of the application. Feel free to ping me once I should take another look at it.

ok . i'm unsure how make the Tables , i can copy an example and edit ?

**Review by Noc2 — CHANGES_REQUESTED** [review]
Sure. See the template: https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md

**Comment by Teke-pixel** [comment]
> Sure. See the template: https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md

how's it looking now ?

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the updates. It looks much better now. I have some follow-up requests: 
- Could you also add some mock-ups/images of the UI to the application or a link to it? 
- For the Democracy & Governance Pallet, are you trying to reuse existing pallets or develop something new? We only support the development of new features/pallets and not the implementation of specific solutions.
- Also, could you take a look at our [guidelines](https://github.com/w3f/Grants-Program#guidelines) and add a comment regarding it or update the application accordingly:
 

> "We do not award grants for projects that have been the object of a successful token sale.
> Applications must not mention a specific token. Furthermore, the focus of the application should lie on the software that is being implemented/research being carried out as part of the grant, and less on your project/venture/operation. For the purpose of the application and delivery, think about how others might also benefit from your work.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. Let me know once I should take another look at it.

**Comment by Teke-pixel** [comment]
> Thanks for the updates. It looks much better now. I have some follow-up requests:
> 
> * Could you also add some mock-ups/images of the UI to the application or a link to it?
> * For the Democracy & Governance Pallet, are you trying to reuse existing pallets or develop something new? We only support the development of new features/pallets and not the implementation of specific solutions.
> * Also, could you take a look at our [guidelines](https://github.com/w3f/Grants-Program#guidelines) and add a comment regarding it or update the application accordingly:
> 
> > "We do not award grants for projects that have been the object of a successful token sale.
> > Applications must not mention a specific token. Furthermore, the focus of the application should lie on the software that is being implemented/research being carried out as part of the grant, and less on your project/venture/operation. For the purpose of the application and delivery, think about how others might also benefit from your work.

i have a link to the build , can i use that instead of a mock-up/image ? image creation might take a while , what do you suggest ? i updated the link in Development status .

We will create a pallet for chessonchain governance , i was trying to point out the similarity to the Democracy pallet , i have changed it to ChessOnchain Governance pallet , its going to be  a different pallet with different features .

i Have also removed token name in the application , although  a token it is part of the project details , project is not aimed at a token sale . let me know if i should remove everything concerning the token .

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the updates. Regarding the token, I would at least remove "CHS Token" and instead simply focus on a generic token or expressly point out as part of the application that the token will be worthless. For the UI, anything that helps us understand what exactly you want to develop would be useful. This could even be a simple drawing. Regarding Democracy & Governance Pallet, it would be nice if you could point out as part of the specification (milestone table) how these are different from the existing "Democracy & Governance Pallets" or what exact functionality they will support.

**Comment by Teke-pixel** [comment]
Ok ✅ cool

File: Chess-On-Chain.md [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+
+| Number |Deliverable  | Specification |
+| -------- | -------- | -------- |
+|0a     | License   | Apache 2.0 / GPLv3 / MIT / Unlicense   |
+| 0b   | Documentation | We will provide a repository with  documentation of the pallets and a basic "How TO Guide " that explains how a user can (for example) spin up one of our Substrate nodes and earn tokens from gameplay , Step by Step guide to spawn up wallets , play chess and Earn tokens          |
+| 0c  |Testing and Testing out  |The code will have unit-test coverage  to ensure functionality . In the guide, we will describe how to run these tests   |
+| 0d   | Docker  :      | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone.    |
+|  0e  | Article :  |  We will write an article on each pallet and build in this milestones    |
+|  1   | OFC Pallet:  | implementation of  offchain workers Pallet to get data from Game played on ChessOnchain As JSON files with http requests  Offchain worker pallet  will collect Game data such as moves , winning streaks ,ranking and rating |
```
Comment (semuelle) [inline Chess-On-Chain.md:207]: This is the first mention of an offchain worker. Can you provide details about what this pallet does, where the data comes from and where it goes?

File: Chess-On-Chain.md [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+| 0b   | Documentation | We will provide a repository with  documentation of the pallets and a basic "How TO Guide " that explains how a user can (for example) spin up one of our Substrate nodes and earn tokens from gameplay , Step by Step guide to spawn up wallets , play chess and Earn tokens          |
+| 0c  |Testing and Testing out  |The code will have unit-test coverage  to ensure functionality . In the guide, we will describe how to run these tests   |
+| 0d   | Docker  :      | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone.    |
+|  0e  | Article :  |  We will write an article on each pallet and build in this milestones    |
+|  1   | Storage Pallet |Implement Storage Pallets  for game Data from offchain workers pallet  to substrate chain  with storage maps. to store for each user/player   |
+|  2.   | Chess Onchain Democracy & Governance Pallet :    |Implememnt and Work on Democracy Pallet for Governance    |
+|  3.   | 	Leader Board Pallet :     |A Decentralized Leaderboard Pallet for competetions and Records  of championship games , and different chess game results     |
```
Comment (semuelle) [inline Chess-On-Chain.md:233]: Please add architecture and/or flow diagrams to show what data and functionality each provides and how the data flows.

**Review by semuelle — CHANGES_REQUESTED** [review]
I would strongly suggest you remove anything token and governance related from this application and just focus on a small proof of concept, ideally with a Level 1 grant. Governance and play-to-earn are nice to have but are not technical novelties or necessary to start this project.

**Comment by semuelle** [comment]
Closed due to inactivity. Let me know if you want to pick it back up and I'll reopen it, @Teke-pixel.

**Comment by Teke-pixel** [comment]
Yes pls open so I will update today , I got a lil ill/slow  @semuelle

**Comment by Teke-pixel** [comment]
> I would strongly suggest you remove anything token and governance related from this application and just focus on a small proof of concept, ideally with a Level 1 grant. Governance and play-to-earn are nice to have but are not technical novelties or necessary to start this project.

i removed token as part of the Deliverables and also removed it in parts of the application , However the token is part of the project details so i have also pointed out that this is not based on a token sale , thanks for the advice , And its true , Play to Earn is not a novelty to start this project , the Governance pallet will not be complicated at first , But some competitions and tournaments will require Democracy , and its in our kick of plan to start simple  tournaments , plus i am applying to get this builds funded , i think it will be smart for my startup to start at them , even if not to the apex ,   Let me know what you think

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. However, it's still unclear to me how the Governance and Democracy pallets work and how they are different from the existing solutions. Also, I find it strange that you want to integrate into a specific wallet as part of your first milestone. Overall, at this point, I'm not willing to support the project, but I will mark it as ready for review so that others can take a look at the application as well.

**Review by takahser — CHANGES_REQUESTED** [review]
@Teke-pixel are you sure you need a custom chain for this use case? I'd suggest to go with a simple smart contract (e.g. ink!) instead. Also, I'm confused how you want to integrate a pallet with a wallet - could you elaborate? Finally, as other W3F grant members have mentioned in this thread, I'm confused on why you wouldn't just reuse existing solutions for governance-related use cases. Feel free to comment on that as well.

**Comment by semuelle** [comment]
Thanks again for your application and the work you put into it, @Teke-pixel. Unfortunately, the committee has decided not to support it. Main reasons are the lack of technical details and very wide scope of the application.

Feel free to apply again in the future. Personally, I think a minimal proof of concept of an on-chain chess game, ideally written in ink! so it can be deployed on existing parachains, would be interesting. 

Let me know if you have any further questions.

**Comment by Teke-pixel** [comment]
I came to this discussion to gather information as we are applying again after so much improvement, thanks guys

**Pull request closed without merging**